src/Integration/Binding/CSharpVBBindingOperation.cs

```diff
@@ -142,6 +142,39 @@ public void Commit()
         #region Helpers
 
         private void AddRuleset()
+        {
+            TrySetNonConditionalRuleSet();
+            SetConditionalRuleSets();
+        }
+
+        /// <summary>
+        /// Try to set a ruleset without specifying a configuration (e.g. Debug / Release).
+        /// No changes are made if the project already has a ruleset.
+        /// </summary>
+        private void TrySetNonConditionalRuleSet()
+        {
+            var hasUserSpecifiedRuleSet = propertyInformationMap.Keys.Any(x => !IsDefaultMicrosoftRuleSet(x.Value as string));
+
+            if (hasUserSpecifiedRuleSet)
+            {
+                // We could've done this for all projects that have configurations that point to the same ruleset but we don't want to create noise for existing users.
+                return;
+            }
+
+            var ruleSetFullFilePath = propertyInformationMap.Values.FirstOrDefault()?.NewRuleSetFilePath;
+
+            if (!string.IsNullOrEmpty(ruleSetFullFilePath))
+            {
+                var relativeRuleSetValue = PathHelper.CalculateRelativePath(ProjectFullPath, ruleSetFullFilePath);
+                var projectSystem = serviceProvider.GetService<IProjectSystemHelper>();
+                projectSystem.SetProjectProperty(initializedProject, Constants.CodeAnalysisRuleSetPropertyKey, relativeRuleSetValue);
+            }
+        }
+
+        /// <summary>
+        /// Set a ruleset for each project configuration (e.g. Debug / Release).
```

Comment:
This needs an additional comment explaining that it will be a no-op for new-style projects that already have an unconditional reference i.e. it's mainly to handle legacy cases of already-bound projects that have conditional ruleset references.